river/InputManager.zig

```diff
@@ -34,6 +37,43 @@ const default_seat_name = "default";
 
 const log = std.log.scoped(.input_manager);
 
+pub const InputDevice = struct {
+    device: *wlr.InputDevice,
+
+    // Careful: The identifier is not unique! A physical input device may have
+    // multiple logical input devices with the exact same vendor id, product id
+    // and name. However identifiers of InputConfigs are unique.
+    identifier: []const u8,
+
+    pub fn init(self: *InputDevice, device: *wlr.InputDevice) !void {
+        // The identifier is formatted exactly as in Sway
+        var identifier = try std.fmt.allocPrint(
+            util.gpa,
+            "{}:{}:{s}",
+            .{ device.vendor, device.product, mem.trim(
+                u8,
+                mem.spanZ(device.name),
```

Comment:
https://github.com/ziglang/zig/pull/8698
```suggestion
                mem.sliceTo(device.name, 0),
```